blitz/ui/components/base.py

```diff
@@ -1,8 +1,60 @@
+from typing import Any, Self, TypeVar
 from blitz.ui.blitz_ui import BlitzUI, get_blitz_ui
 
+T = TypeVar("T", bound="BaseComponent")
 
-class BaseComponent:
-    def __init__(self, blitz_ui: BlitzUI = get_blitz_ui()) -> None:
-        self.blitz_ui: BlitzUI = blitz_ui
-        self.current_project = blitz_ui.current_project
-        self.current_app = blitz_ui.current_app
+
+# Get the blitz_ui through a metaclass
+class BaseComponentMeta(type):
+    def __new__(cls, name: str, bases: tuple[type, ...], namespace: dict[str, Any]) -> type:
+        blitz_ui = get_blitz_ui()
+        namespace["blitz_ui"] = blitz_ui
```

Comment:
Remove Useless variable
```suggestion
        namespace["blitz_ui"] = get_blitz_ui()
```